### PR TITLE
Proxy or Mirror Support for download

### PR DESCRIPTION
**Pull request by soiamsoNG**
Want to add proxy support considering below point

Amazon AWS with very slow connection around 30K/s in some area.
VSCode not support socks5 proxy in a clean way.
Node not support socks5 proxy in a clean way.

Hope can solve the AWS very slow issue in another way but not set a proxy.

### CODE REVIEW

**Comment by soiamsoNG** [comment]
c53475b, let user can set a mirror path in setting

**Comment by soiamsoNG** [comment]
https://github.com/haskell/vscode-haskell/pull/309/commits/40d866ff7e8d4f5086b77bc7645f05b963457ec7
for case like set mirror path to below json, which will issue a redirect
https://github.com/soiamsoNG/haskell-language-server-via-fastgit/releases/download/0.0.1/releasesInfo.json

**Comment by jneira** [comment]
@soiamsoNG only to be sure: the actual download mechanism does not honour the standard system proxy settings (`http_proxy`, `https_proxy`, etc)?

**Comment by soiamsoNG** [comment]
> @soiamsoNG only to be sure: the actual download mechanism does not honour the standard system proxy settings (`http_proxy`, `https_proxy`, etc)?

Thanks for your remind.

**Comment by soiamsoNG** [comment]
sorry for mistake press the button

**Comment by jneira** [comment]
> 
> 
> sorry for mistake press the button

So, could you use those env variables to tell the extension the proxy to be used? If that is the case, maybe we should add to documentation, instead add extra code.

File: package.json [hunk 1]
```diff
@@ -146,6 +146,18 @@
...
+          "scope": "resource",
+          "type": "string",
+          "default": "https://api.github.com/repos/haskell/haskell-language-server/releases",
+          "description": "a path to a json file"
+        },
+        "haskell.languageServerDownloadProxy": {
+          "scope": "resource",
+          "type": "string",
+          "default": "",
+          "description": "http:// https:// socks:// socks5:// pac+http://"
```
Comment (bubba) [inline package.json:159]: If we are using the vscode http proxy setting, do we need to have our own proxy configuration in this extension too? I think we can just remove this one then

File: package.json [hunk 2]
```diff
@@ -146,6 +146,18 @@
           "default": "haskell-language-server",
           "description": "Which language server to use."
         },
+        "haskell.languageServerReleasesPath": {
+          "scope": "resource",
+          "type": "string",
+          "default": "https://api.github.com/repos/haskell/haskell-language-server/releases",
+          "description": "a path to a json file"
```
Comment (bubba) [inline package.json:153]: Nitpick
```suggestion
          "description": "An optional path to override where to check for haskell-language-server releases"
```

File: package.json [hunk 3]
```diff
@@ -146,6 +146,18 @@
           "default": "haskell-language-server",
           "description": "Which language server to use."
         },
+        "haskell.languageServerReleasesPath": {
+          "scope": "resource",
+          "type": "string",
+          "default": "https://api.github.com/repos/haskell/haskell-language-server/releases",
```
Comment (bubba) [inline package.json:152]: Can we make the default "", and then inside the extension use "https://api.github.com/repos/haskell/haskell-language-server/releases" if it is set to ""

File: package.json (hunk 1 above)
Comment (soiamsoNG) [reply package.json:159]: a little bit too much, please don't pick the proxy part

Comment (soiamsoNG) [reply package.json:159]: I create #313 for just Mirror Support part

**Comment by bubba** [comment]
Sorry for the late reply on this: Is this still needed? Or does the built-in vscode proxy work ok?

**Comment by soiamsoNG** [comment]
No need now

**Pull request closed without merging**